include/navigation.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file navigation.hpp
+ * @author Hritvik Choudhari (hac@umd.edu)
+ * @brief Declaration file for the Navigation class, responsible for generating the search path and moving the robot.
+ * @version 0.1
+ * @date 2023-12-09
+ * 
+ * @copyright Copyright (c) 2023
+ * 
+ */
+#pragma once
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/pose.hpp"
+#include "./Perception.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+
+#include <memory>
+#include <iostream>
+#include <vector>
+#include <chrono>
+
+using POSE = geometry_msgs::msg::PoseStamped;
```

Comment:
Don't use using and using namespaces. Instead write it each time so that we are sure of the type of variable and message that is passed to and returned by the functions, topics, services and actions